roles/KubevirtNodeLabeller/tasks/main.yml

```diff
@@ -4,15 +4,98 @@
     state: present
     definition: "{{ item | from_yaml }}"
   with_items: "{{ lookup('template', 'kubevirt-node-labeller-roles.yaml.j2').split('\n---\n') | select('search', '(^|\n)[^#]') | list }}"
+
 - name: Create the node labeller daemon set
   k8s:
     state: present
     definition: "{{ lookup('template', 'kubevirt-node-labeller-ds.yaml.j2') | from_yaml }}"
   register: nl
-- name: "Wait for the node-labeller to start"
+
+- name: "refresh node-labeller var"
+  k8s:
+    state: present
+    definition: "{{ lookup('k8s', kind=nl.result.kind, namespace=nl.result.metadata.namespace, resource_name=nl.result.metadata.name) | from_yaml }}"
+  register: nl_status
```

Comment:
it seems to me here you're fetching the data in a convoluted way. Wouldn't a `set_fact` from lookup work?
If not, why?